### DISCUSSION TITLE
[DeepSpeed] Multi-GPU can't converge

### DISCUSSION THREAD

**Discussion by thomas-happify** [Lightning-AI/pytorch-lightning#8019] [DDP / multi-GPU / multi-node]
```
seed_everything(43, worker=True)

Trainer(gpus=1, accumulate_grad_batches=16, accelerator='ddp', plugin='deepspeed',  max_epoch=20)
Trainer(gpus=2, accumulate_grad_batches=8, accelerator='ddp', plugin='deepspeed', max_epoch=20)
```

shouldn't these two args have similar training results? 
when I use `gpus=2` and `accumulate_grad_batches=8`, the model can't converge.

**Comment by ethanwharris** [comment]
@thomas-happify Thanks for your question :smiley: Are you able to provide a reproducible example showing this in practice? Without that it may be difficult to say why the performances are different.

**Reply by SeanNaren** [reply]
Hey @thomas-happify I've updaetd the issue tracker here: https://github.com/PyTorchLightning/pytorch-lightning/issues/8058#issuecomment-886611562

seems it's working as intended, but let me know if there is anything missing from the reproduce.

**Comment by tchaton** [comment]
Dear @thomas-happify,

Mind providing a reproducible script using the BoringModel. I believe those 2 approaches should be almost the same at the difference of dropped last batches from the DistributedSampler.

Best,
T.C
